internal/providers/detect.go

```diff
@@ -155,7 +156,14 @@ func isTerraformDir(path string) bool {
 	return terraform.IsTerraformDir(path)
 }
 
+// goformation lib is not threadsafe, so we run this check synchronously
+// See: https://github.com/awslabs/goformation/issues/363
+var cfMux = &sync.RWMutex{}
```

Comment:
Again not sure if we need a read/write mutex. Would prefer just simple mutex to avoid confusion.

```suggestion
var cfMux = &sync.Mutex{}
```